src/storage/src/hummock/vector/writer/hnsw.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2025 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::mem::take;
+
+use bytes::{Bytes, BytesMut};
+use prost::Message;
+use rand::SeedableRng;
+use rand::rngs::StdRng;
+use risingwave_hummock_sdk::HummockObjectId;
+use risingwave_hummock_sdk::vector_index::{
+    HnswFlatIndex, HnswFlatIndexAdd, HnswGraphFileInfo, VectorFileInfo, VectorStoreInfoDelta,
+};
+use risingwave_pb::hummock::PbHnswGraph;
+
+use crate::dispatch_measurement;
+use crate::hummock::vector::file::VectorFileBuilder;
+use crate::hummock::vector::writer::{VectorObjectIdManagerRef, new_vector_file_builder};
+use crate::hummock::vector::{EnumVectorAccessor, get_vector_block};
+use crate::hummock::{HummockResult, SstableStoreRef};
+use crate::opts::StorageOpts;
+use crate::store::Vector;
+use crate::vector::DistanceMeasurement;
+use crate::vector::hnsw::{
+    HnswBuilderOptions, HnswGraphBuilder, VectorAccessor, VectorStore, insert_graph, new_node,
+};
+
+struct HnswVectorStore {
```

Comment:
IIUC, `HnswVectorStore` is used by hnsw vector writer only to retrieve vevctor for the flushed / building vectors in the current epoch while `FileVectorStore` is used by hnsw vector reader only to retrieve vector for the checkpointed vectors. I found this a little bit confusing:
- We should differentiate them in the naming.
- Similar to hummock local/global state store, I think the executor needs to have a unified view of both in order to retrive the correct result. What is the mechanism of doing that for vector?